svelte-kit-scss/src/lib/components/FileExplorer/models.ts

```diff
@@ -0,0 +1,12 @@
+import type { GithubRepoContentsItemType } from '$lib/constants/github';
```

Comment:
Honest question, how are we distinguishing `models` vs `interfaces`?

 - `src/lib/components/FileExplorer/models.ts`
 - `src/lib/interfaces`

Reply:
Personally I like the idea of splitting it into 3 things, each with their own interfaces:

 - **getting the data** (from an API and verifying with zod or something for types)
 - **the data** (or state or store) which is the types the front-end application "has" or expects
 - **showing the data** (the components) which optimally DON'T use the types from the store/state, they just have their own "props" interfaces and they base everything off of that
 
 ...and none of these 3 layers (...or systems, or modules or whatever to call them) share types
 
 **THAT BEING SAID**: I think for these demo apps, the 3 split might be a bit much, but we could go for just a split of:
 
- interfaces used by functions/classes (Services?) for getting the data 
- mapping functions from 1 type to another
- components having their own type (the models here)

So maybe we should put the types in `src/lib/interfaces` in with the things that fetch data?
(This may be a bit of a rabbit hole/sidetrack that doesn't need to be resolved this PR or maybe not even in this project)